pkg/api/openshiftclusterdocument.go

```diff
@@ -50,3 +52,15 @@ type OpenShiftClusterDocument struct {
 func (c *OpenShiftClusterDocument) String() string {
 	return encodeJSON(c)
 }
+
+func (doc *OpenShiftClusterDocument) ImageRegistryStorageAccountName() (string, error) {
+	if doc == nil {
+		return "", fmt.Errorf("OpenShift cluster document is nil")
+	}
```

Comment:
Here it is the same. no need to check if it is nil, since it checked in `pkg/backend/openshiftcluster.go:48`  already

Reply:
If I remove the condition, I will need to check  m.doc != nil on `pkg/cluster/deploystorage.go:161` before calling this method. For me it makes no sense to traverse the call stack to **see** if at that point m.doc is nil or not, I want to **ensure** it is not nil). 

The good think about doing it inside the method is that we don't ever need to do in the codebase:
```go
if doc != nil {
    err := doc.ImageRegistryStorageAccountName()
}
``` 
we can simply do 
```go
err :=  doc.ImageRegistryStorageAccountName()
```
and forget about the panic case.

If I don't check doc == nil inside the method I will do it just before every call to doc.ImageRegistry(), here is just one call but I am talking about the general idea of the check inside the method. 

I will remove the nil check from inside the method and add the nil check before calling imageRegistryStorageAccountName() because I still can not convince you 😂👌